src/parsing.jl

```diff
@@ -257,12 +257,31 @@ end
 
 # ISO, YMD
 _DEFAULT_TYPE_MAP[:timestamptz] = ZonedDateTime
-const TIMESTAMPTZ_FORMATS = (
-    dateformat"y-m-d HH:MM:SSz",
-    dateformat"y-m-d HH:MM:SS.sz",
-    dateformat"y-m-d HH:MM:SS.ssz",
-    dateformat"y-m-d HH:MM:SS.sssz",
+const TIMESTAMPTZ_FORMATS = Dict(
+    ZonedDateTime => (
+        dateformat"y-m-d HH:MM:SSz",
+        dateformat"y-m-d HH:MM:SS.sz",
+        dateformat"y-m-d HH:MM:SS.ssz",
+        dateformat"y-m-d HH:MM:SS.sssz",
+    ),
+    UTCDateTime => (
+        dateformat"y-m-d HH:MM:SS",
+        dateformat"y-m-d HH:MM:SS.s",
+        dateformat"y-m-d HH:MM:SS.ss",
+        dateformat"y-m-d HH:MM:SS.sss",
+    ),
 )
+
+function _pqparse(::Type{T}, str::AbstractString) where T<:Union{UTCDateTime, ZonedDateTime}
+    formats = TIMESTAMPTZ_FORMATS[T]
```

Comment:
Is this dynamic lookup adding time? Should this be a function generated in a loop with `@eval` instead?

Reply:
So just pushing the lookup to the dispatch system?

Reply:
Okay, looks like making it separate tuple consts with a dispatch results in a moderate improvement.

```
julia> @benchmark LibPQ.pqparse(ZonedDateTime, "2004-10-19 10:23:54-00")
BenchmarkTools.Trial: 10000 samples with 10 evaluations.
 Range (min … max):  1.217 μs … 165.042 μs  ┊ GC (min … max): 0.00% … 97.32%
 Time  (median):     1.288 μs               ┊ GC (median):    0.00%
 Time  (mean ± σ):   1.365 μs ±   3.119 μs  ┊ GC (mean ± σ):  4.49% ±  1.96%

         ▃▆ ▃▇▇██▆▅▇▅ ▆▄▄▄▁▁▁
  ▂▂▃▄▅▇███▆█████████▁█████████▁█▇▇▆▅▆▅▅▅▁▄▅▄▄▄▃▃▃▃▁▃▃▃▃▃▂▂▂▂ ▅
  1.22 μs         Histogram: frequency by time        1.44 μs <

 Memory estimate: 1.98 KiB, allocs estimate: 16.

 julia> @benchmark LibPQ.pqparse(UTCDateTime, "2004-10-19 10:23:54-00")
BenchmarkTools.Trial: 10000 samples with 66 evaluations.
 Range (min … max):  851.000 ns …  11.239 μs  ┊ GC (min … max): 0.00% … 91.27%
 Time  (median):     869.318 ns               ┊ GC (median):    0.00%
 Time  (mean ± σ):   894.653 ns ± 473.587 ns  ┊ GC (mean ± σ):  2.51% ±  4.36%

            ▂▃█▃▂▁
  ▂▂▂▂▃▃▅▅▇████████▇▇▅▄▃▃▃▃▃▂▂▂▂▃▂▂▂▂▂▂▂▂▂▂▂▂▂▂▂▂▂▂▂▂▂▂▁▂▂▂▂▂▂▂ ▃
  851 ns           Histogram: frequency by time          934 ns <

 Memory estimate: 1.58 KiB, allocs estimate: 9.
```

 Dispatch:

```
julia> @benchmark LibPQ.pqparse(ZonedDateTime, "2004-10-19 10:23:54-00")
BenchmarkTools.Trial: 10000 samples with 10 evaluations.
 Range (min … max):  1.079 μs … 276.938 μs  ┊ GC (min … max): 0.00% … 99.56%
 Time  (median):     1.100 μs               ┊ GC (median):    0.00%
 Time  (mean ± σ):   1.135 μs ±   2.758 μs  ┊ GC (mean ± σ):  2.43% ±  1.00%

       ▃ █  ▄
  ▂▁▄▁▁█▁█▁▁█▁█▁▁▇▁▆▄▁▇▁▁▆▁▆▁▁▅▁▄▁▁▄▁▃▁▁▃▁▁▃▁▃▁▁▃▁▂▁▁▂▁▂▁▁▂▁▂ ▃
  1.08 μs         Histogram: frequency by time        1.18 μs <

 Memory estimate: 784 bytes, allocs estimate: 13.

julia> @benchmark LibPQ.pqparse(UTCDateTime, "2004-10-19 10:23:54-00")
BenchmarkTools.Trial: 10000 samples with 155 evaluations.
 Range (min … max):  678.768 ns …   9.551 μs  ┊ GC (min … max): 0.00% … 90.86%
 Time  (median):     690.858 ns               ┊ GC (median):    0.00%
 Time  (mean ± σ):   710.735 ns ± 300.160 ns  ┊ GC (mean ± σ):  1.60% ±  3.52%

     ▄█▄▃▁▁
  ▂▃████████▆▇▅▅▃▃▃▄▄▄▄▃▄▄▄▄▄▄▃▃▂▂▂▂▂▂▂▂▃▃▃▃▃▃▂▂▂▂▂▂▂▂▂▁▂▂▂▂▂▁▂ ▃
  679 ns           Histogram: frequency by time          768 ns <

 Memory estimate: 464 bytes, allocs estimate: 6.
```